### PR TITLE
:warning: tvk-cleanup krew plugin changes for 2.6.x releases

### PR DESCRIPTION
**Pull request by sachin-trilio**
Minor changes for tvk-cleanup krew plugin for 2.6.x releases

### CODE REVIEW

File: tools/cleanup/cleanup.sh [hunk 1]
```diff
@@ -69,24 +69,14 @@ delete_tvk_op() {
...
   fi
 
   # For Upstream OR in case if TVK installed on OCP using "helm"
   # Delete Triliovault-manager and Triliovault-operator using helm/label
   if (helm list -A | grep -v REVISION | grep triliovault >/dev/null 2>&1); then
     echo "Uninstalling Trilivault-manager"
-    tvm=$(helm list -A | grep -v REVISION | grep triliovault-v | awk '{print $1}')
-    tvm_ns=$(helm list -A | grep -v REVISION | grep triliovault-v | awk '{print $2}')
+    tvm=$(helm list -A | grep -v REVISION | grep 'triliovault-[0-9]' | awk '{print $1}')
+    tvm_ns=$(helm list -A | grep -v REVISION | grep 'triliovault-[0-9]' | awk '{print $2}')
```
Comment (msurudwad) [inline tools/cleanup/cleanup.sh:79]: - Both `tvm` and `tvm_ns` will contain a list of helm releases if multiple releases are installed in cluster. You'll have to handle the list with loop.
- `helm list -A | grep -v REVISION | grep 'triliovault-[0-9]' ` is common in both commands, store the result in variable and do `awk` on it

File: tools/cleanup/cleanup.sh [hunk 2]
```diff
@@ -105,6 +95,19 @@ delete_tvk_op() {
       fi
     fi
   fi
+
+  # Delete k8s-triliovault-resource-cleaner cronjob
+  tvkcron=$(kubectl get cronjob --no-headers -A 2>/dev/null | grep k8s-triliovault | awk '{print $2}')
+  tvkcron_ns=$(kubectl get cronjob --no-headers -A 2>/dev/null | grep k8s-triliovault | awk '{print $1}')
```
Comment (msurudwad) [inline tools/cleanup/cleanup.sh:101]: same here `kubectl get cronjob --no-headers -A 2>/dev/null | grep k8s-triliovault` is common in both commands

File: tools/cleanup/cleanup.sh [hunk 3]
```diff
@@ -105,6 +95,19 @@ delete_tvk_op() {
...
+  if [ -n "${tvkcron}" ]; then
+    echo "Deleting k8s-triliovault-resource-cleaner cronjob"
```
Comment (msurudwad) [inline tools/cleanup/cleanup.sh:103]: add `ns` name in log

**Review by msurudwad — CHANGES_REQUESTED** [review]

File: tools/cleanup/cleanup.sh (hunk 1 above)
Comment (sachin-trilio) [reply tools/cleanup/cleanup.sh]: Done

File: tools/cleanup/cleanup.sh (hunk 2 above)
Comment (sachin-trilio) [reply tools/cleanup/cleanup.sh]: Done

File: tools/cleanup/cleanup.sh (hunk 3 above)
Comment (sachin-trilio) [reply tools/cleanup/cleanup.sh]: Done

File: tools/cleanup/cleanup.sh [hunk 4]
```diff
@@ -73,39 +73,55 @@ delete_tvk_op() {
...
+  tvm_ns=$(helm list -A | grep -v REVISION | grep 'triliovault' | awk '{print $2}' | uniq)
+  if [ -n "${tvm_ns}" ]; then
+    for ns in ${tvm_ns}; do
+
+      # Deleting Trilivault-manager CR
+      tvm_name=$(kubectl get triliovaultmanager --no-headers -n "${ns}" | awk '{print $1}')
+      echo "Patching triliovaultmanager CR ${tvm_name} in namespace ${ns}"
+      kubectl patch triliovaultmanager "${tvm_name}" -p '{"metadata":{"finalizers":[]}}' --type=merge -n "${ns}"
+      echo "Deleting triliovaultmanager CR ${tvm_name} in namespace ${ns}"
+      kubectl delete triliovaultmanager "${tvm_name}" -n "${ns}"
```
Comment (msurudwad) [inline tools/cleanup/cleanup.sh:86]: for all `kubectl delete` commands add ` --force --grace-period=0 --timeout=5s` [if timeout exceeded command will return non-zero exit code, handle this]

first run `kubectl delete` then, run `kubectl patch`. Do this at all the places in this script.

File: tools/cleanup/cleanup.sh [hunk 5]
```diff
@@ -69,42 +69,61 @@ delete_tvk_op() {
...
-    tvo=$(helm list -A | grep -v REVISION | grep triliovault-o | awk '{print $1}')
-    tvo_ns=$(helm list -A | grep -v REVISION | grep triliovault-o | awk '{print $2}')
-    if [ -n "${tvo}" ]; then
-      helm uninstall "${tvo}" -n "${tvo_ns}"
-      retValue=$?
-      if [ "${retValue}" -ne 0 ]; then
-        exit_status=1
+
+      # Uninstall Trilivault-manager
+      tvm=$(helm list -n "${ns}" | grep -v REVISION | grep 'triliovault-[0-9]' | awk '{print $1}')
```
Comment (msurudwad) [inline tools/cleanup/cleanup.sh:93]: `tvm` can contain multiple values, need to handle this loop as well

**Review by msurudwad — CHANGES_REQUESTED** [review]

File: tools/cleanup/cleanup.sh (hunk 1 above)
Comment (msurudwad) [reply tools/cleanup/cleanup.sh]: handling for `tvm` not done

File: tools/cleanup/cleanup.sh (hunk 4 above)
Comment (sachin-trilio) [reply tools/cleanup/cleanup.sh]: done

File: tools/cleanup/cleanup.sh (hunk 5 above)
Comment (sachin-trilio) [reply tools/cleanup/cleanup.sh:103]: Not addressed as discussed - k8s-triliovault  SA name will have clash in same ns..

**Review by msurudwad — APPROVED** [review]

**Pull request merged**